rosie.py

```diff
@@ -1,49 +1,64 @@
-from sys import argv, exit
+"""Rosie.
 
+Usage:
+  rosie.py run (chamber_of_deputies | federal_senate)
+  rosie.py run (chamber_of_deputies | federal_senate) [--years <years>]
+  rosie.py run (chamber_of_deputies | federal_senate) [--path <path>]
+  rosie.py run (chamber_of_deputies | federal_senate) [--path] [--years <years>]
+  rosie.py test [chamber_of_deputies | federal_senate]
 
-def entered_command(argv):
-    if len(argv) >= 2:
-        return argv[1]
-    return None
+Options:
+    --years   runs Rosie fetching data and finding suspicions for a specific set of years
+    --path    directory in which Rosie saves fetched data and the suspicions CSV [default: /tmp/serenata-data]
 
+"""
 
-def help():
-    message = (
-        'Usage:',
-        '  python rosie.py run chamber_of_deputies [<path to output directory>]',
-        'Testing:',
-        '  python rosie.py test',
-        '  python rosie.py test chamber_of_deputies',
-    )
-    print('\n'.join(message))
+from sys import exit
+from docopt import docopt
 
 
-def run():
+def main():
+    arguments = docopt(__doc__)
+
+    if arguments['run']:
+        run(**arguments)
+
+    if arguments['test']:
+        return test(**arguments)
+
+
+def run(**args):
     import rosie
     import rosie.chamber_of_deputies
     import rosie.federal_senate
 
-    if len(argv) >= 3:
-        target_module = argv[2]
-    else:
-        print('A module must be provided.')
-        help()
-        exit(1)
-    target_directory = argv[3] if len(argv) >= 4 else '/tmp/serenata-data/'
+    target_directory = args['<path>'] if args['--path'] else '/tmp/serenata-data/'
+
+    commands = ('chamber_of_deputies', 'federal_senate')
+    target_module, *_ =  filter(lambda x: arguments.get(x), commands)
+
     klass = getattr(rosie, target_module)
-    klass.main(target_directory)
 
+    if args['--years']:
+        years = args['<years>']
+        years = [int(num) for num in years.split(',')]
+        klass.main(target_directory, years=years)
+    else:
+        klass.main(target_directory)
 
-def test():
+
+def test(**args):
     import os
 
     import unittest
 
     loader = unittest.TestLoader()
 
-    if len(argv) >= 3:
-        target_module = argv[2]
-        tests_path = os.path.join('rosie', target_module)
+    if args['chamber_of_deputies']:
+        tests_path = os.path.join('rosie', 'chamber_of_deputies')
+        tests = loader.discover(tests_path)
+    elif args['federal_senate']:
+        tests_path = os.path.join('rosie', 'federal_senate')
         tests = loader.discover(tests_path)
     else:
         tests = loader.discover('rosie')
```

Comment:
Minor DRY suggestion. What about:

```python
test_path = 'rosie'
if args['chamber_of_deputies']:
     tests_path = os.path.join(test_path, 'chamber_of_deputies')
elif args['federal_senate']:
     tests_path = os.path.join(test_path, 'federal_senate')
tests = loader.discover(test_path)
```

Reply:
I still think that this is not good enough :/
maybe another way to test this with `@patch`?